test/functional/forall/reduce-basic/tests/test-forall-basic-ReduceBitAnd.hpp

```diff
@@ -9,20 +9,25 @@
 #define __TEST_FORALL_BASIC_REDUCEBITAND_HPP__
 
 #include <cstdlib>
+#include <ctime>
 #include <numeric>
+#include <vector>
 
-template <typename IDX_TYPE, typename DATA_TYPE, typename WORKING_RES, 
+template <typename IDX_TYPE, typename DATA_TYPE,
+          typename SEG_TYPE,
           typename EXEC_POLICY, typename REDUCE_POLICY>
-void ForallReduceBitAndBasicTestImpl(IDX_TYPE first, IDX_TYPE last)
+void ForallReduceBitAndBasicTestImpl(const SEG_TYPE& seg,
+                                     const std::vector<IDX_TYPE>& seg_idx,
+                                     camp::resources::Resource& working_res)
 {
-  RAJA::TypedRangeSegment<IDX_TYPE> r1(first, last);
+  IDX_TYPE data_len = seg_idx[seg_idx.size() - 1] + 1;
```

Comment:
Nice, this is a clever way to get these tests to work with all the segment types. 👍